app/[userID]/page.tsx

```diff
@@ -0,0 +1,214 @@
+"use client";
+import { ThemeToggle } from "@/components/theme-trigger";
+import { Alert, AlertDescription, AlertTitle } from "@/components/ui/alert";
+import { Button } from "@/components/ui/button";
+import { MessageTypes, useServerConnection } from "@/hooks/useServerConnection";
+import { Info } from "lucide-react";
+
+import {
+    Card,
+    CardContent,
+    CardFooter,
+    CardHeader,
+    CardTitle,
+} from "@/components/ui/card";
+
+import { Verified } from "lucide-react";
+import Link from "next/link";
+import { redirect } from "next/navigation";
+import { useState } from "react";
+
+export default function Home({ params }: { params: { userID: string } }) {
+    const userId = params.userID;
+    const [userInfo, setUserInfo] = useState();
+    let acceptedInfo = false;
+    const { socket, onMessageType } = useServerConnection();
+    onMessageType(MessageTypes.FailedIdentification, () => {
+        if (!acceptedInfo) {
+            acceptedInfo = true;
+            setUserInfo(false);
+        }
+    });
+    onMessageType(MessageTypes.Identification, (info) => {
+        if (!acceptedInfo) {
+            acceptedInfo = true;
+            setUserInfo(info);
+        }
+    });
+    socket?.addEventListener(
```

Comment:
These should be put inside useEffect so that they don't run during server-side rendering, and also two listeners don't attach to "open" when the component is re-rendered

Reply:
When I attempt to use useEffect, the function doesn't run. The check is there because it will return an error otherwise.

Reply:
Alright I see